docs: add macOS Internet Sharing and Parallels Desktop conflicts for Boundary Client Agent (#5496)
hashicorp/boundary: website/content/docs/api-clients/client-agent.mdx

If you run Docker Desktop 4.26 or later, you must clear the `Use kernel networking for UDP` option.
Otherwise, the Client Agent refuses to start.

> ### Internet Sharing (macOS)

Reviewer: Whichever casing is correct, we should be consistent. We use `MacOS` elsewhere in this document, so please us `MacOS` when describing this OS.

Author: Updated.

Enabling Internet Sharing on macOS causes the system's DNS resolver (`mDNSResponder`) to bind to all interfaces on port 53, preventing the Client Agent from starting. Disable Internet Sharing before running the Client Agent.

### Parallels Desktop (macOS VM on macOS Host)

> Running a macOS VM in Parallels Desktop with shared networking causes the system's DNS resolver (`mDNSResponder`) to bind to all interfaces, blocking port 53 and preventing the Client Agent from starting.

Reviewer: Could we add a short workaround here too? "Configure Parallels Desktop with direct network" or whatever the workaround is, I can't remember 😁.

Author: Added a workaround.

If you run Docker Desktop 4.26 or later, you must clear the `Use kernel networking for UDP` option.
Otherwise, the Client Agent refuses to start.

### Internet Sharing (MacOS)

> Enabling Internet Sharing on MacOS causes the system's DNS resolver (`mDNSResponder`) to bind to all interfaces on port 53, preventing the Client Agent from starting. Disable Internet Sharing before running the Client Agent.

Reviewer 2: ```suggestion
Enabling **Internet Sharing** on MacOS causes the system's DNS resolver (`mDNSResponder`) to bind to all interfaces on port 53, preventing the Client Agent from starting. Disable **Internet Sharing** before you run the Client Agent.
```

### Parallels Desktop (MacOS VM on MacOS Host)
Running a MacOS VM in Parallels Desktop with shared networking causes the system's DNS resolver (`mDNSResponder`) to bind to all interfaces, blocking port 53 and preventing the Client Agent from starting.

> **Workaround:** Configure Parallels Desktop to use a different network mode.
> Go to **Hardware > Network > Source**, and select an alternative network configuration such as **Host-Only, Default Adapter, or Wi-Fi** instead of Shared Network.

Reviewer 2: ```suggestion
As a workaround, you can configure Parallels Desktop to use a different network mode.
Select an alternative network configuration such as **Host-Only**, **Default Adapter**, or **Wi-Fi** instead of **Shared Network**.
Refer to the Parallels Desktop documentation for more information.
```

Reviewer 2: I'd suggest that we refer users to the Parallels documentation instead of providing explicit steps for configuring this, in case they change it in a future version.
